boa3_test/tests/compiler_tests/test_file_generation.py

```diff
@@ -776,3 +776,321 @@ def fact(self, f: int) -> int:
         if f <= 1:
             return 1
         return f * self.fact(f - 1)
+
+    def test_generate_manifest_file_with_type_hint_list(self):
```

Comment:
I assume this is where you are checking if the manifest is generated correctly. I am gonna be honest that my untrained eyes can't understand it. But if you guys think this is fine, its fine by me.